stores/filestore.go

```diff
@@ -2041,19 +2468,16 @@ func (ms *FileMsgStore) backgroundTasks() {
 			ms.Lock()
 			opened := 0
 			for _, slice := range ms.files {
-				if slice.file != nil || slice.idxFile != nil {
-					opened++
-					if slice.lastUsed < timeTick && time.Duration(timeTick-slice.lastUsed) >= time.Second {
-						if slice.file != nil {
-							slice.file.Close()
-							slice.file = nil
-						}
-						if slice.idxFile != nil {
-							slice.idxFile.Close()
-							slice.idxFile = nil
-						}
-						opened--
-					}
+				// If no FD limit and this is the write slice, skip.
+				if !ms.hasFDsLimit && slice == ms.writeSlice {
```

Comment:
The line you refer to optimizes the fact that there is no need to notify (through the setting of the atomic variable) the background task that slices should be checked if the given slice is the write slice. The code here is simply saying that when check is happening, we should not close the writeSlice if there is no FD limit.